### ISSUE TITLE
sql: FOREIGN KEY constraint failed on same table

### ISSUE THREAD

**Issue by dezdoz** [tarantool/tarantool#4661]
Tarantool version:
2.3.0-249-gf6775e8

OS version:
Ubuntu 19.10

Bug description:
Cannot update records which have foreign key constraints on the same table. SQLite is working fine.

Steps to reproduce:
`CREATE TABLE IF NOT EXISTS "t1" (
		"id" NUMBER NOT NULL PRIMARY KEY,
		"pid" NUMBER,
		FOREIGN KEY ("pid") REFERENCES "t1"("id")
);`

`INSERT INTO "t1" ("id", "pid") VALUES (1, 2);`

`UPDATE "t1" SET "pid" = 3 WHERE "id" = 1;`

Returned error on _update_: **FOREIGN KEY constraint failed**

**Comment by Korablev77** [comment]
Hello. Insert operation obviously violates foreign key constraint. SQLite doesn't throw an error since foreign keys are disabled by default. To enable them one should use `"PRAGMA foreign_keys = ON;"`. With this pragma insert fails in SQLite as well.

**Comment by dezdoz** [comment]
Hey, sorry I've posted a wrong sql example. This one is correct and also working in sqlite but not in tarantool.

`
PRAGMA foreign_keys = ON;
`

`CREATE TABLE IF NOT EXISTS "t1" ( "id" NUMBER NOT NULL PRIMARY KEY, "pid" NUMBER, FOREIGN KEY ("pid") REFERENCES "t1"("id") );`

`
INSERT INTO "t1" ("id") VALUES (1);
`

`
INSERT INTO "t1" ("id") VALUES (2);
`

`
INSERT INTO "t1" ("id", "pid") VALUES (3, 1);
`

`
UPDATE "t1" SET "pid" = 3 WHERE "id" = 2;
`

**Comment by Korablev77** [comment]
Ok, this example really looks wrong. Looks like a bug.

**Comment by victor-oancea-leftclick-io** [comment]
@slumber @kyukhin @Korablev77 Hey guys, I have the exact situation mentioned above. And I'm using latest release version  (2.10.x)

Here's my scenario:
```
CREATE TABLE dummy (
    pk INTEGER PRIMARY KEY,
    fk INTEGER NOT NULL REFERENCES dummy (pk));
INSERT INTO dummy VALUES (1, 1), (2, 1), (3,2);
UPDATE dummy SET fk = 1 WHERE pk = 3; ===> `Failed to execute SQL statement: FOREIGN KEY constraint failed`
``` 

I see some commits to fix this from 2+ years ago, what happened? Are there any plans to fix this?

**Comment by ImeevMA** [comment]
After closing #6986 this issue has been fixed for version 2.11. For version 2.10 and earlier, the issue will most likely not be fixed.
